aten/src/ATen/NestedTensorImpl.h

```diff
@@ -32,6 +32,9 @@ struct TORCH_API NestedTensorImpl : public c10::TensorImpl {
   const Tensor& get_nested_size_tensor() const {
     return nested_size_tensor_;
   }
+  c10::optional<int64_t> get_opt_size(int64_t i) const {
```

Comment:
We could make this lazy and only calculate and store once the method was called.

Reply:
I assume the description is "analogous to `Tensor::size()`, but only applies if all sub-tensors (or however you want to word it) have the same size in the ith dimension". Would help to write that. Also, I think it should be called `opt_size` to match `Tensor::size()`?